File: SenLib/Sen3/FileFixes/t_jump.cs
using HyoutaUtils;
using HyoutaUtils.Streams;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SenLib.Sen3.FileFixes {
	class t_jump : FileMod {
		public string GetDescription() {
			return "Fix minor error in fast-travel menu text.";
		}

		public IEnumerable<FileModResult> TryApply(FileStorage storage) {
			var file_en = storage.TryGetDuplicate(new HyoutaUtils.Checksum.SHA1(0x57dd3ac63f52d21bul, 0x677d5e1e7c20a7f2ul, 0x55d0689fu));
			if (file_en == null) {
				return null;
			}

			var s = file_en.CopyToMemory();
			s.SwapBytes(0x1b7d, 0x1b7d + 1);

			return new FileModResult[] {
				new FileModResult("data/text/dat_en/t_jump.tbl", s),
			};
		}

		public IEnumerable<FileModResult> TryRevert(FileStorage storage) {
			var file_en = storage.TryGetDuplicate(new HyoutaUtils.Checksum.SHA1(0x57dd3ac63f52d21bul, 0x677d5e1e7c20a7f2ul, 0x55d0689fu));
			if (file_en == null) {
				return null;
			}
			return new FileModResult[] {
				new FileModResult("data/text/dat_en/t_jump.tbl", file_en.Duplicate()),
			};
		}
	}
}

File: native/sen3/FileFixes/t_jump.cpp
#include <filesystem>
#include <string_view>
#include <vector>

#include "../file_fixes.h"

#include "../../p3a/pack.h"
#include "../../p3a/structs.h"
#include "../../sha1.h"

namespace SenLib::Sen3::FileFixes::t_jump {
std::string_view GetDescription() {
    return "Fix minor error in fast-travel menu text.";
}

bool TryApply(const SenPatcher::GetCheckedFileCallback& getCheckedFile,
              std::vector<SenPatcher::P3APackFile>& result) {
    try {
        auto file = getCheckedFile(
            "data/text/dat_en/t_jump.tbl",
            22928,
            SenPatcher::SHA1FromHexString("57dd3ac63f52d21b677d5e1e7c20a7f255d0689f"));
        if (!file) {
            return false;
        }

        auto& bin = file->Data;
        std::swap(bin[0x1b7d], bin[0x1b7d + 1]);

        result.emplace_back(SenPatcher::P3APackFile{
            std::move(bin), file->Filename, SenPatcher::P3ACompressionType::LZ4});

        return true;
    } catch (...) {
        return false;
    }
}
} // namespace SenLib::Sen3::FileFixes::t_jump
